src/timer.c

```diff
@@ -619,17 +641,32 @@ void rt_soft_timer_check(void)
 
             /* remove timer from timer list firstly */
             _rt_timer_remove(t);
+            if (!(t->parent.flag & RT_TIMER_FLAG_PERIODIC))
+            {
+                t->parent.flag &= ~RT_TIMER_FLAG_ACTIVATED;
+            }
+            /* add timer to temporary list  */
+            rt_list_insert_after(&list, &(t->row[RT_TIMER_SKIP_LIST_LEVEL - 1]));
```

Comment:
Miss rt_list_remove(&(t->row[RT_TIMER_SKIP_LIST_LEVEL - 1])); ?
We got the hard fault after invoke rt_timer_check in pm.c

Reply:
thanks